### PR TITLE
SKILLCAPES

### PR DESCRIPTION
**Pull request by Takahiru**
### Adds skillcapes

It adds skillcapes! You get a regular skillcape if you have over 300 hours in a job, a trimmed skillcape at 500 hours. There are special skillcapes for mentors/admins/senioradmins. If you have 72 unlocked capes (all trimmed job capes worth) then you get the mighty maximum cape! The option will be in preferences under game preferences.

Thanks to war0100/tex for the sprites

Also it's a bounty i like bounty

#### Changelog

:cl:  
rscadd: Skillcapes
imageadd: Tex's skillcapes
/:cl:

### CODE REVIEW

**Comment by Mqiib** [comment]
time to show off how good I am at greytiding WOOOOOOOOOOO

**Comment by patpol4** [comment]
Is there a observer skillcape

**Comment by Takahiru** [comment]
> Is there a observer skillcape

no

**Comment by Partheo** [comment]
Not a fan of adding capes to a spaceman game. Also not a fan of meta rewards.

**Comment by Marmio64** [comment]
mentor cape time

**Comment by ynot01** [comment]
sprites are kind of cringe (at a glance, might look better ingame idk) but i like the idea

**Comment by Takahiru** [comment]
ingame ex
heres an ingame example

**Comment by Moja10** [comment]
What IC explanation is there for someone to have a cape for being an admin, mentor, or senior admin?

**Comment by Takahiru** [comment]
> What IC explanation is there for someone to have a cape for being an admin, mentor, or senior admin?

the names arent explicitly "mentor cape" "admin cape" on purpose.
edit-- mentor is "cape of the sage", admin is "cape of mighty judgement", SA is "cape of the supreme judge".
I could tone it done further to make the names less 'epic' if you wish.

**Comment by Mqiib** [comment]
https://forums.yogstation.net/threads/10-skillcapes-finish-what-was-started-ss13-server.22046/

There was a thread on the forums for a while where y'all could have voiced your general complaints about the theme of this pr and the IC/OOC-type issues.

**Comment by Tipy1802** [comment]
I both like it and don't like it.
While I do like fancy clothing, this seems way out of place and off putting.
It's also very metagamey. People being able to pick who is ''good'' at a glance sounds bad, yeah true they can already do that if they know that [insert name here] is a good doctor but that takes more work. 
I don't want heads being fired because the captain saw someone else had a skillcape. Or lizards being promoted to head because ''he has cape''.
I also don't feel like every job should get a cape. Like no assistants shouldn't get fancy capes or being treated more highly than even the captain because of some OOC factor.

I would like it more if they were like some offical clothing for heads of staff and not some far too fancy 24/7 meta clothing no matter the IC status of someone

**Comment by Xoxeyos** [comment]
I don't like it, we're trying to go towards a more MRP orientated environment, and I believe interactions based on **skill** rather than **communication** would reflect poorly on what we are trying to achieve, I do also believe that these sprites really don't look the greatest. Also **Administration orientated capes would reflect poorly on our administration team**, as it would potentially make players believe that we actually gain something from the capes, and that they are out balancing those that do not have them, even if it isn't true.

**Comment by Takahiru** [comment]
If a consensus is come to i will change the PR, I added the admin/mentor capes because its part of the original bounty.

**Review by Partheo — CHANGES_REQUESTED** [review]
The only way I'd be even slightly down with this is if they were something that made sense IC, like badges, cloaks, or hoodies instead of capes.

**Comment by Moja10** [comment]
Small badges would be a lot better, I agree with Partheo

**Comment by Takahiru** [comment]
since moja approves, i will eventually change it, spriting is hard.

**Comment by MegaEmpirical** [comment]
Badges is much nicer

**Comment by ynot01** [comment]
the host approved bounty was for skill**capes**, with the following criteria met

**Comment by Takahiru** [comment]
> the host approved bounty was for skill**capes**, with the following criteria met
> 

i forgot about the mentor/admin skillcapes being autodrobe only, but its hard to track greentexts so i didnt do that
other than those i fulfilled the original bounty

**Comment by Partheo** [comment]
I'm not sure what the procedure is if there's an approved bounty that the headcoders disagree with. I don't think this has happened before. Maybe worth a council vote to establish a procedure?

That being said. I'm willing to help with the respriting if it is decided that that's the route we're taking.

**Comment by MegaEmpirical** [comment]
Whilst I am vastly more supportive of badges, I'd hate to see all the time making capes go to waste.

I'd like to see badges used instead for how this PR us suposed to work but maybe make the capes a department item? Or occasional spawn?

**Comment by Takahiru** [comment]
Adding onto partheo, i dont get what should be done under the circumstances of changing a bounty, id make some suggested changes if the reason i made this pr wasnt for the bounty

**Comment by swissloaf** [comment]
skillcapes good, badges suck

File: code/__HELPERS/global_lists.dm [hunk 1]
```diff
@@ -53,6 +53,10 @@
 		GLOB.materials_list[D.id] = D
 
 	GLOB.emote_list = init_emote_list()
+	//Skillcapes
+	for(var/path in typecacheof(/obj/item/clothing/neck/skillcape))
```
Comment (Bibby0110) [inline code/__HELPERS/global_lists.dm]: Am I reading this right? You're making a skillcape in nullspace just to store it in a GLOB? Why don't you juse store the path?

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
ok this hurts i'm not going to review it

File: code/__HELPERS/global_lists.dm (hunk 1 above)
Comment (Takahiru) [reply code/__HELPERS/global_lists.dm]: because its easier to access the variables e

Comment (Bibby0110) [reply code/__HELPERS/global_lists.dm]: what variables could you possibly need to access

Comment (Takahiru) [reply code/__HELPERS/global_lists.dm]: a datum item?

**Comment by Moggram** [comment]
>removing the B& hammer because its OOC in IC.
>Add OOC meta skillcape

Pick one.

**Comment by Reed0506** [comment]
bruh if a captain demoted a head in favor of some guy with a cape theyd be banned from playing head roles

thats the way i see it at least

**Comment by maxion12345** [comment]
I'd love to see these encouraging people to play different jobs to get them fucin capes
Achievements was like, supposed to help but nobody gives a shit about those

**Comment by Takahiru** [comment]
The PR is now completely in line with the original bounty. The mentor/admin skillcapes have been moved to the autodrobe rather than being only for admins/mentors. Xantam also said the bounty will be nulled if there are any changes, partheeeooo

**Review by Partheo — APPROVED** [review]
I don't like this PR

**Comment by Takahiru** [comment]
After this gets merged we can try and shift it over to badges, all it will take is a bunch of respriting.

**Comment by MegaEmpirical** [comment]
I like the spirit of it and think badges would have been fantastic as the idea of 50 people running around with capes is terrifying. Still, solid work.

**Comment by Mqiib** [comment]
iirc there aren't actually THAT many people with significantly high playtimes to get a cape. I don't remember when but someone posted an image of all the people that had 200+ hours in a role and it really wasn't that long of a list.

**Comment by Takahiru** [comment]
> iirc there aren't actually THAT many people with significantly high playtimes to get a cape. I don't remember when but someone posted an image of all the people that had 200+ hours in a role and it really wasn't that long of a list.

yeah most people cant stand to play the same job for 300 hours straight.

**Review by jcatjmeow — APPROVED** [review]
Lookin good

I like par also am not a fan of meta rewards or the idea of wearing a cape in general...but I'm not strongly against it, and the code is solid and everything

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -1606,6 +1609,28 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+
+					var/pickedskillcape = input(user, "Choose your Skillcape.", "Character Preference", skillcape) as null|anything in selectablecapes
+					if(!selectablecapes.len)
+						to_chat(user, "You have no availiable skillcapes!")
+						return
+					var/count = 1
+					for(var/A in GLOB.skillcapes)
+						if(A == pickedskillcape)
+							break
+						count = count+1
```
Comment (Amelia0010) [inline code/modules/client/preferences.dm]:
```suggestion
						count++
```

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -1606,6 +1609,28 @@ GLOBAL_LIST_EMPTY(preferences_datums)
...
+						if(user.client.prefs.exp[A.job] >= A.hours)
+							if(!A.special)
+								selectablecapes += A	
+						if(A.special) //check for special capes
+							if(A.capetype == "max")
+								if(selectablecapes.len >= 72) //72 is the amount of job skillcapes, including trimmed.
+									selectablecapes += A
+
+					var/pickedskillcape = input(user, "Choose your Skillcape.", "Character Preference", skillcape) as null|anything in selectablecapes
+					if(!selectablecapes.len)
```
Comment (Amelia0010) [inline code/modules/client/preferences.dm:1622]: this should be moved above the input.

**Review by Amelia0010 — CHANGES_REQUESTED** [review]

**Review by Amelia0010 — APPROVED** [review]

**Comment by ThatGuyPhil** [comment]
oh boy time to play ss13 for 4 years nonstop to get all these

**Comment by Cartlord** [comment]
And the badges never came about.

**Pull request merged**